src/IO/BufferBase.h

```diff
@@ -59,42 +59,46 @@ class BufferBase
     BufferBase(Position ptr, size_t size, size_t offset)
         : pos(ptr + offset), working_buffer(ptr, ptr + size), internal_buffer(ptr, ptr + size) {}
 
-    void set(Position ptr, size_t size, size_t offset)
+    virtual ~BufferBase() {}
+
+    virtual void set(Position ptr, size_t size, size_t offset)
```

Comment:
Is it intended? I'm convinced that making every method virtual is extremely bad idea. For now our buffers are already too complex, but there are several invariants and we somehow can live with them.

Reply:
I made method virtual because need a decorator class for buffer.
I'll think about other decision here.